liteflow-labs/starter-kit
# Generalize creation of NFT for any collection

### Project organization

- Dependant on Release with https://github.com/liteflow-labs/liteflow-js/pull/102

### Description

This PR uses the next version of the hooks to generalize the creation of NFT to any collection (compatible).
The URL to create an NFT is now changed and includes the `chainId` and `collectionAddress`.
The application now defines the list of collections open for mint in the environment. If this list is empty, the "create" button in the menu is automatically removed.

### How to test

Install the hook library from https://github.com/liteflow-labs/liteflow-js/pull/102 with `npm i path-to-hook-package`, then go to the create page.
You should not see anything.
Now add in the environment `NEXT_PUBLIC_MINTABLE_COLLECTIONS=0x...,0x...` based on the data available in the API you use, and you should be able to see one tile for each collection. 
Test to mint an new NFT in all the different configurations:
- ERC721 with mint
- ERC1155 with mint
- ERC721 with lazymint
- ERC1155 with lazymint
All should work with no issue

### TODO:
- [x] Update library when released (current issue with the build)
- [x] Update documentation with the new envs https://github.com/liteflow-labs/liteflow-js/pull/112
